### PR TITLE
/poll command: 發起投票，且一定時間後結束投票並展示投票結果

### PR DESCRIPTION
**Pull request by Johnny850807**
## Why need this change? / Root cause: 
- #98 
## Changes made:
- when use poll command, bot will respond the embedded message with polling options 
- work in progress (toggle: off)
## Test Scope / Change impact:
- `/poll`

### Summary by OpenAI

Release Notes:
- New Feature: Added `/poll` command to the bot, allowing users to initiate and participate in polling sessions.
- New Feature: Added functionality to observe and remove members from a list periodically.
- New Feature: Added a new subcommand for muting audiences and fixed a logical error in the `isNotMuteCommand()` function.
- Bug fix: Modified input validation functions to retrieve options from a Discord slash command interaction event with validation.
- Chore: Added a new dependency to the project and included changes to implement the `/poll` command that responds with an embedded message containing polling options.

> "New features abound,
> Polls and observers all around,
> Bugs and errors have been found,
> With this PR, our code is sound."

### CODE REVIEW

**Review by frankvicky — APPROVED** [review]
LGTM

File: poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        }?.let { TimeUnit.valueOf(it.uppercase()) } ?: return null
+
+        val question = getOption(OPTION_QUESTION)!!.asString
+        val options = getOption(OPTION_OPTIONS)!!.asString.split(Regex("\\s*,\\s*"))
+        if (options.size > EMOJI_UNICODES.size) {
+            reply("The number of options cannot be greater than ${EMOJI_UNICODES.size}.").complete()
+            return null
+        }
+
+        return PollingSetting(time!!, timeUnit, question, options)
```
Comment (Wally5077) [inline poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt:92]: 改在宣告處使用 `!!`

File: poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    override fun onMessageReactionAdd(event: MessageReactionAddEvent) {
+        with(event) {
+            val session = sessionIdToSession[messageId] ?: return
+            if (userId == jda.selfUser.id) {
```
Comment (Wally5077) [inline poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt:98]: NIT：在 `MessageReactionAddEvent` 和 `MessageReactionRemoveEvent` 的父類開一個回傳 boolean 的擴充方法

File: poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                    .complete()
+        }
+    }
+}
+
+
+data class PollingSetting(val time: Long, val timeUnit: TimeUnit, val question: String, val options: List<String>) {
+    private val optionsMessageBody = options.mapIndexed { i, option ->
+        "${EMOJI_UNICODES[i]} $option"
+    }.joinToString("\n")
```
Comment (Wally5077) [inline poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt:133]: 1. `System.lineSeparator()`
2. static import

File: poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                            .build(),
+                    EmbedBuilder()
+                            .setTitle("Options")
+                            .setDescription(optionsMessageBody)
+                            .setColor(Color.RED)
+                            .build(),
+                    EmbedBuilder()
+                            .setTitle("Settings")
+                            .setDescription("Session ends in $time $timeUnit.")
+                            .setColor(Color.BLUE)
```
Comment (Wally5077) [inline poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt:151]: `static import`

File: poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+class PollingResult(private val voterIdToVotedOptionIndices: Map<String, MutableList<Int>>, private val setting: PollingSetting) {
+    val messageBody: String
+        get() {
+            return voterIdToVotedOptionIndices.values
+                    .flatten()
+                    .groupingBy { it }
+                    .eachCount()
+                    .map { (index, count) -> "${setting.getOption(index)}: $count votes." }
+                    .joinToString("\n")
```
Comment (Wally5077) [inline poll/src/main/kotlin/tw.waterballsa.utopia.poll/PollCommandListener.kt:198]: ditto `System.lineSeparator()`

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done.

**Review by Wally5077 — APPROVED** [review]
LGTM

**Pull request merged**